File: checkout-vat-java/src/test/java/de/metro/evaluate/interview/ProcessorTest.java
package de.metro.evaluate.interview;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Map;
import org.junit.jupiter.api.Test;

public class ProcessorTest {

    @Test
    void process() {
        //given
        var prices = Map.of("pepsi", 10.0);
        var articleVats = Map.of("pepsi", 15);
        //when
        var processor = new Processor();
        var report = processor.process(prices, articleVats);

        //then
        assertEquals(report.total, 11.5);
    }
}

File: checkout-vat/src/test/kotlin/de/metro/evaluate/interview/ProcessorTest.kt
package de.metro.evaluate.interview

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class ProcessorTest {

    @Test
    fun process() {
        //given
        val prices = mapOf("pepsi" to 10.0)
        val articleVats = mapOf("pepsi" to 15)
        //when
        val processor = Processor()
        val report = processor.process(prices, articleVats)

        //then
        assertEquals(report.total, 11.5)
    }
}